strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/cron/jobs/MergeMavenGroupRepositoryIndexCronJob.java

```diff
@@ -0,0 +1,75 @@
+package org.carlspring.strongbox.cron.jobs;
+
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.cron.domain.CronTaskConfigurationDto;
+import org.carlspring.strongbox.cron.jobs.fields.*;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexFounder;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexFounder.RepositoryIndexFounderQualifier;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryTypeEnum;
+
+import javax.inject.Inject;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class MergeMavenGroupRepositoryIndexCronJob
```

Comment:
Clarification: So, how will this work? Each group repository will have one of these automatically created when the repository is created, or what?

Reply:
Group index will be merged from the underlying sub-repositories of this group.

> Each group repository will have one of these automatically created when the repository is created, or what?

Yes, by default. See `MavenRepositoryManagementStrategy#createRepositoryInternal`